gemspec: Update emoji_regex version requirement to 1.0

rye opened the pull request:
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context

`emoji_regex` is out of date, and this update did not break anything. It's used in `pilot/lib/pilot/build_manager.rb` and I've verified that the API hasn't changed.

### Description

I ran `bundle install` and then tested.

rye committed:
gemspec: Update emoji_regex version requirement to 1.0

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

joshdholtz commented on fastlane.gemspec:
```diff
@@ -97,7 +97,7 @@ Gem::Specification.new do |spec|
   spec.add_dependency('google-cloud-storage', '>= 1.15.0', '< 2.0.0') # Access Google Cloud Storage for match
 
-  spec.add_dependency('emoji_regex', '~> 0.1') # Used to scan for Emoji in the changelog
+  spec.add_dependency('emoji_regex', '~> 1.0') # Used to scan for Emoji in the changelog
```
```suggestion
  spec.add_dependency('emoji_regex', '>= 0.1' , '< 2.0') # Used to scan for Emoji in the changelog
```

Can we actually change it to be a `>=` and `<=`? This gives us more control over the versions in the future and won't force anybody to bump their `emoji_regex` versions (run `bundle install`) if they aren't even using these feature in pilot 🙃

joshdholtz reviewed (changes requested):
Small request!

Commit:
gemspec: Use explicit version range for emoji_regex

At the suggestion of @joshdholtz.

I'm not sure how I feel about this instead, since it goes against RubyGems' own suggestions, but this is consistent with the rest of this file.

Co-Authored-By: rye <kristofer.rye@gmail.com>

rye commented:
(I used the "Apply suggestion" feature.)

joshdholtz commented:
@rye Thanks! I'll take care of the lint error 😝

Commit:
Rubo stuff

joshdholtz reviewed (approved):
We good now! Thanks for the bump here 💪
